core/modules/file/file.pages.inc

```diff
@@ -1,10 +1,389 @@
+      if ($new_file_uri = file_unmanaged_copy($replacement->uri, $destination_uri, $replace_mode)) {
+
+        // Rename the temporary file to avoid confusion.
+        $replacement->filename = $file->fid . '-replacement-' . $replacement->filename;
+        $replacement->save();
+
+        // Remove temporary file.
+        file_delete($replacement);
+
+        // Update if the uri target has changed.
+        if ($new_file_uri != $file->uri) {
+          // Store the original file uri to delete if save is successful.
+          $orphaned_uri = $file->uri;
+
+          // Update file entity uri.
+          $file->uri = $new_file_uri;
+        }
+      }
+    }
+
+    // Clear any image styles for this file.
+      // @todo check and see if it's an image once we have $file->type.
+    $image_info = image_get_info($file->uri);
+    if (!empty($image_info)) {
+      image_path_flush($file->uri);
+    }
+  }
+
+  // Run entity form submit handling and save the file.
+  entity_form_submit_build_entity('file', $file, $form, $form_state);
+
+  // A user might assign the associated user by entering a user name in the file
+  // edit form, which we then need to translate to a user ID.
+  if (isset($file->name)) {
+    // The use of isset() is mandatory in the context of user IDs, because
+    // user ID 0 denotes the anonymous user.
+    if ($user = user_load_by_name($file->name)) {
+      $file->uid = $user->uid;
+    }
+    else {
+      // Anonymous user.
+      $file->uid = 0;
+    }
+  }
+  elseif ($file->uid) {
+    $user = user_load($file->uid);
+    $file->name = $user->name;
+  }
+
+  if (file_uri_scheme($file->uri) != $form_state['values']['scheme']) {
+    $file_destination = $form_state['values']['scheme'] . '://' . file_uri_target($file->uri);
+    $file_destination = file_stream_wrapper_uri_normalize($file_destination);
+    $file_destination_dirname = backdrop_dirname($file_destination);
+    // Create the directory in case it doesn't exist.
+    file_prepare_directory($file_destination_dirname, FILE_CREATE_DIRECTORY);
+    if ($moved_file = file_move($file, $file_destination, FILE_EXISTS_RENAME)) {
+      // Only re-assign the file object if file_move() did not fail.
+      $file = $moved_file;
+    }
+  }
+
+  file_save($file);
+
+  $args = array(
+    '%title' => entity_label('file', $file),
+  );
+  watchdog('file', 'File: updated %title.', $args);
+  backdrop_set_message(t('File: %title has been updated.', $args));
+
+  // Clean up orphaned file.
+  if (!empty($orphaned_uri)) {
+    file_unmanaged_delete($orphaned_uri);
+
+    $args['@orphaned'] = file_uri_target($orphaned_uri);
+    watchdog('file', 'File: deleted orphaned file @orphaned for %title.', $args);
+    backdrop_set_message(t('The replaced @type @orphaned has been deleted.', $args));
+  }
+
+  $form_state['redirect'] = 'admin/content/files';
+
+  // Clear the page and block caches.
+  cache_clear_all();
```

Comment:
We do not call `cache_clear_all()` in Backdrop on content changes (which would clear the entire page cache). Should be removed.